includes/helper/class-wp-job-manager-helper-listing-renewal.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Helper_Listing_Renewal.
+ *
+ * @package wp-job-manager
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Handles the renewal of Job Listings.
+ *
+ * @since $next-version$$
+ */
+class WP_Job_Manager_Helper_Listing_Renewal {
+
+	/**
+	 * Submit job form instance.
+	 *
+	 * @var WP_Job_Manager_Form_Submit_Job
+	 */
+	public $form = null;
+
+	/**
+	 * Stores static instance of class.
+	 *
+	 * @access protected
+	 * @var WP_Job_Manager_Form_Submit_Job The single instance of the class
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param WP_Job_Manager_Form_Submit_Job $form Form object.
+	 */
+	public function __construct( WP_Job_Manager_Form_Submit_Job $form ) {
+		$this->form = $form;
+
+		if ( $this->is_renew_action() ) {
+			add_filter( 'submit_job_steps', [ $this, 'remove_edit_steps_for_renewal' ] );
+			add_filter( 'submit_job_step_preview_submit_text', [ $this, 'submit_button_text_renewal' ], 15 );
+		}
+	}
+
+	/**
+	 * Returns static instance of class.
+	 *
+	 * @param WP_Job_Manager_Form_Submit_Job $form Form object.
+	 * @return self
+	 */
+	public static function instance( WP_Job_Manager_Form_Submit_Job $form ) {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self( $form );
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Checks if 'renew' is set as action.
+	 *
+	 * @since $$next-version$$
+	 * @return bool
+	 */
+	public function is_renew_action() {
+		$job_id = isset( $_GET['job_id'] ) ? sanitize_text_field( wp_unslash( $_GET['job_id'] ) ) : '';
+		$nonce  = isset( $_GET['nonce'] ) ? wp_unslash( $_GET['nonce'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- Nonce should not be modified.
+		$action = 'job_manager_renew_job_' . $job_id;
+		if ( ! wp_verify_nonce( $nonce, $action ) ) {
+			return false;
+		}
+		return isset( $_GET['action'] ) && sanitize_text_field( wp_unslash( $_GET['action'] ) ) === 'renew';
+	}
+
+	/**
+	 * Handle steps for renewing a listing before expiry, removes edit steps and overrides preview step.
+	 *
+	 * @access private
+	 * @param array $steps
```

Comment:
Missing the description

Reply:
https://github.com/Automattic/WP-Job-Manager/pull/2422/commits/2eb330f440bcd4bf0dbb1d1d54de5306448ba041